jmclient/jmclient/auth.py

```diff
@@ -0,0 +1,85 @@
+import datetime
+import os
+from typing import Optional
+
+import jwt
+
+from jmbase.support import bintohex
+
+
+class InvalidScopeError(Exception):
+    pass
+
+
+def get_random_key(size: int = 16) -> str:
+    """Create a random key has an hexadecimal string."""
+    return bintohex(os.urandom(size))
+
+
+class JMTokenAuthority:
+    """Manage authorization tokens."""
+
+    SESSION_VALIDITY = {
+        "access": datetime.timedelta(minutes=30),
+        "refresh": datetime.timedelta(hours=4),
+    }
+    SIGNATURE_ALGORITHM = "HS256"
+
+    def __init__(self, wallet_name: Optional[str] = None):
+        self.signature_key = {
+            "access": get_random_key(),
+            "refresh": get_random_key(),
+        }
+        self.wallet_name = wallet_name
+
+    def verify(
+        self,
+        token: str,
+        scope: str = "walletrpc",
+        *,
+        is_refresh: bool = False,
+        verify_exp: bool = True,
+    ):
+        """Verify JWT token.
+
+        Token must have a valid signature and its scope must contain both scopes in
+        arguments and wallet_name property.
+        """
+        token_type = "refresh" if is_refresh else "access"
+        claims = jwt.decode(
```

Comment:
Afaict from looking at the pyjwt code, this will throw `ExpiredSignatureError` if we passed the expiration. I think there's a value in catching that to bubble up an error message for the caller? So they can distinguish between a rejected token that is just invalid vs one that is expired?

Sometimes there's an argument in computer security for never telling the caller the reason for the failure, since it can allow probing attacks. That doesn't seem to apply here.